libraries/shared/factories/event/repository_test.go

```diff
@@ -0,0 +1,211 @@
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package event_test
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/vulcanize/vulcanizedb/libraries/shared/factories/event"
+	"github.com/vulcanize/vulcanizedb/libraries/shared/test_data"
+	"github.com/vulcanize/vulcanizedb/pkg/datastore/postgres"
+	"github.com/vulcanize/vulcanizedb/pkg/datastore/postgres/repositories"
+	"github.com/vulcanize/vulcanizedb/pkg/fakes"
+	"github.com/vulcanize/vulcanizedb/test_config"
+	"math/big"
+)
+
+var _ = Describe("Repository", func() {
+	var db *postgres.DB
+
+	BeforeEach(func() {
+		db = test_config.NewTestDB(test_config.NewTestNode())
+		test_config.CleanTestDB(db)
+	})
+
+	Describe("Create", func() {
+		const createTestEventTableQuery = `CREATE TABLE public.testEvent(
+		id        SERIAL PRIMARY KEY,
+		header_id INTEGER NOT NULL REFERENCES headers (id) ON DELETE CASCADE,
+		log_id    BIGINT  NOT NULL REFERENCES header_sync_logs (id) ON DELETE CASCADE,
+		variable1 TEXT,
+		UNIQUE (header_id, log_id)
+		);`
+		const addCheckedColumnQuery = `ALTER TABLE public.checked_headers
+		ADD COLUMN testevent INTEGER NOT NULL DEFAULT 0;`
+
+		var (
+			headerID, logID  int64
+			headerRepository repositories.HeaderRepository
+			testModel        event.InsertionModel
+		)
+
+		BeforeEach(func() {
+			db = test_config.NewTestDB(test_config.NewTestNode())
+			test_config.CleanTestDB(db)
+		})
+
+		BeforeEach(func() {
+			_, _ = db.Exec(createTestEventTableQuery)
+			_, _ = db.Exec(addCheckedColumnQuery)
+			headerRepository = repositories.NewHeaderRepository(db)
+			var insertHeaderErr error
+			headerID, insertHeaderErr = headerRepository.CreateOrUpdateHeader(fakes.FakeHeader)
+			Expect(insertHeaderErr).NotTo(HaveOccurred())
+			headerSyncLog := test_data.CreateTestLog(headerID, db)
+			logID = headerSyncLog.ID
+
+			testModel = event.InsertionModel{
+				SchemaName: "public",
+				TableName:  "testEvent",
+				OrderedColumns: []event.ColumnName{
+					event.HeaderFK, event.LogFK, "variable1",
+				},
+				ColumnValues: event.ColumnValues{
+					event.HeaderFK: headerID,
+					event.LogFK:    logID,
+					"variable1":    "value1",
+				},
+			}
+		})
+
+		AfterEach(func() {
+			db.MustExec(`DROP TABLE public.testEvent;`)
+			db.MustExec(`ALTER TABLE public.checked_headers DROP COLUMN testevent;`)
```

Comment:
Why are we dealing with `checked_headers` here? Thought we didn't need that anymore